syne_tune/search_space.py

```diff
@@ -14,956 +14,66 @@
-
-    """
-    def __init__(self, lower: float, upper: float, size: int,
-                 log_scale: bool = False, cast_int: bool = False):
-        assert lower < upper
-        assert size >= 2
-        if log_scale:
-            assert lower > 0.0
-        self._uniform_int = randint(0, size - 1)
-        self.lower = lower
-        self.upper = upper
-        self.log_scale = log_scale
-        self.cast_int = cast_int
-        self.size = size
-        if not log_scale:
-            self._lower_internal = lower
-            self._step_internal = (upper - lower) / (size - 1)
-        else:
-            self._lower_internal = np.log(lower)
-            upper_internal = np.log(upper)
-            self._step_internal = \
-                (upper_internal - self._lower_internal) / (size - 1)
-
-    def _map_from_int(self, x: int) -> Union[float, int]:
-        y = x * self._step_internal + self._lower_internal
-        if self.log_scale:
-            y = np.exp(y)
-        y = np.clip(y, self.lower, self.upper)
-        if not self.cast_int:
-            return float(y)
-        else:
-            return int(np.round(y))
-
-    @property
-    def value_type(self):
-        return float if not self.cast_int else int
-
-    def _map_to_int(self, value) -> int:
-        int_value = np.clip(value, self.lower, self.upper)
-        if self.log_scale:
-            int_value = np.log(int_value)
-        sz = len(self._uniform_int)
-        return int(np.clip(round(
-            (int_value - self._lower_internal) / self._step_internal),
-            0, sz - 1))
-
-    def cast(self, value):
-        return self._map_from_int(self._map_to_int(value))
-
-    def set_sampler(self, sampler, allow_override=False):
-        raise NotImplementedError()
-
-    def get_sampler(self):
-        return None
-
-    def sample(self, spec=None, size=1, random_state=None):
-        int_sample = self._uniform_int.sample(spec, size, random_state)
-        if size > 1:
-            return [self._map_from_int(x) for x in int_sample]
-        else:
-            return self._map_from_int(int_sample)
-
-    @property
-    def domain_str(self):
-        return f"({self.lower}, {self.upper}, {self.__len__()})"
-
-    def __len__(self):
-        return len(self._uniform_int)
-
-    def match_string(self, value) -> str:
-        return str(self._map_to_int(value))
-
-    def __eq__(self, other) -> bool:
-        return isinstance(other, FiniteRange) \
-               and np.isclose(self.lower, other.lower) \
-               and np.isclose(self.upper, other.upper) \
-               and self.log_scale == other.log_scale \
-               and self.cast_int == other.cast_int
-
-
-def sample_from(func: Callable[[Dict], Any]):
-    """Specify that tune should sample configuration values from this function.
-
-    Arguments:
-        func: An callable function to draw a sample from.
-    """
-    return Function(func)
+def _deprecated_warning(name: str):
+    logger.warning(
```

Comment:
When will we remove this deprecation? We should include when support will be removed also I think, it could be in the next version for instance.